wls-broadcast-service/src/main/java/de/muenchen/oss/wahllokalsystem/broadcastservice/service/BroadcastService.java

```diff
@@ -0,0 +1,93 @@
+        this.broadcastMapper = broadcastMapper;
+    }
+
+    @Value("${service.info.oid}")
+    private String serviceOid;
+
+    @PreAuthorize("hasAuthority('Broadcast_BUSINESSACTION_Broadcast')")
+    public void broadcast(final BroadcastMessageDTO messageToBroadcast) {
+        log.debug("#broadcast");
+
+        if (null == messageToBroadcast || null == messageToBroadcast.wahlbezirkIDs() || messageToBroadcast.wahlbezirkIDs().size() <= 0
+                || StringUtils.isEmpty(messageToBroadcast.nachricht()) || StringUtils.isBlank(messageToBroadcast.nachricht())) {
+            throw FachlicheWlsException.withCode(BroadcastExceptionKonstanten.CODE_NACHRICHTENABRUFEN_PARAMETER_UNVOLLSTAENDIG).inService(serviceOid)
+                    .buildWithMessage("Das Object BroadcastMessage ist nicht vollständig.");
+        }
+
+        List<Message> messagesToSave = broadcastMapper.fromBroadcastMessageDTOtoListOfMessages(messageToBroadcast, LocalDateTime.now());
+
+        messageRepo.saveAll(messagesToSave);
+    }
+
+    @PreAuthorize("hasAuthority('Broadcast_BUSINESSACTION_GetMessage')")
+    public MessageDTO getOldestMessage(String wahlbezirkID) throws FachlicheWlsException {
+        log.debug("#nachrichtenAbrufen wahlbezirkID {} length {}", wahlbezirkID, wahlbezirkID.length());
+
+        if (StringUtils.isEmpty(wahlbezirkID) || StringUtils.isBlank(wahlbezirkID)) {
+            throw FachlicheWlsException.withCode(BroadcastExceptionKonstanten.CODE_NACHRICHTENABRUFEN_PARAMETER_UNVOLLSTAENDIG).inService(serviceOid)
+                    .buildWithMessage("wahlbezirkID is blank or empty");
+        }
+
+        val message = messageRepo.findFirstByWahlbezirkIDOrderByEmpfangsZeit(wahlbezirkID);
+
+        if (message.isEmpty()) {
+            throw FachlicheWlsException.withCode(ExceptionKonstanten.CODE_ENTITY_NOT_FOUND).inService(serviceOid).buildWithMessage("No message found");
+        }
+
+        return broadcastMapper.toDto(message.get());
+    }
+
+    @PreAuthorize("hasAuthority('Broadcast_BUSINESSACTION_MessageRead')")
+    public void deleteMessage(String nachrichtID) { //TODO UUID als Parameter
+        if (StringUtils.isEmpty(nachrichtID) || StringUtils.isBlank(nachrichtID)) {
+            throw FachlicheWlsException.withCode(BroadcastExceptionKonstanten.CODE_NACHRICHTENABRUFEN_PARAMETER_UNVOLLSTAENDIG)
+                    .buildWithMessage("nachrichtID is blank or empty");
+        }
+
+        try {
+            UUID nachrichtUUID;
+            nachrichtUUID = java.util.UUID.fromString(nachrichtID);
+            messageRepo.deleteById(nachrichtUUID);
+        } catch (Exception e) {
+            if (e instanceof NumberFormatException) {
+                throw FachlicheWlsException.withCode(BroadcastExceptionKonstanten.CODE_NACHRICHTENABRUFEN_PARAMETER_UNVOLLSTAENDIG).inService(serviceOid)
+                        .buildWithMessage("Nachricht-UUID bad format");
+            } else {
+                throw e;
+            }
+        }
```

Comment:
- Zeile 80 und 81 hätten zusammengefasst werden sollen
- wenn du nur eine NumberFormatException catch willst gib das bitte im Catch-Block an
- `fromString` liefert im Fehlerfall eine `IllegalArgumentException`
- die Exception wurde im Altsystem nicht explizit gehandelt, daher ist es eher Verbesserungspotential

Reply:
Ich dachte ich catche erstmal alles für alle Eventualitäten und wenn NumberFormatException ist, dann diese in FachlicheWlsException umwandeln. Aber nicht, dass andere unbehandelt bleiben.

Reply:
resolved 3e7d275a3127470f107ae6421a5b9867d493f346